test/java/seedu/addressbook/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,146 @@
+
+        // addressbook has more people than displayList
+        // This can happen when view from filtered list caused by some commands(eg. FindCommand)
+        assertViewSuccess(typicalAddressBook, listWithSome, 1);
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * invalid index.
+     */
+    private void assertViewErrorInvalidIndex(AddressBook addressBook, List<ReadOnlyPerson> list,
+                                             int targetVisibleIndex) {
+        assertViewError(addressBook, list, targetVisibleIndex, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * person not existing in the addressbook.
+     */
+    private void assertViewErrorPersonNotInAddressBook(AddressBook addressBook, List<ReadOnlyPerson> list,
+                                                       int targetVisibleIndex) {
+        assertViewError(addressBook, list, targetVisibleIndex, Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
+    }
+
+    /**
+     * Asserts that both a ViewCommand and a ViewAllCommand can retrieve from
+     * the {@code addressBook} details of the person at the given {@code index} in the given {@code list}.
+     *
+     * @param targetVisibleIndex one-indexed position of the target person in the list
+     */
+    private void assertViewSuccess(AddressBook addressBook, List<ReadOnlyPerson> relevantPersons,
+                                   int targetVisibleIndex) {
+        ReadOnlyPerson personToBeViewed = relevantPersons.get(targetVisibleIndex - 1); // -1 is because index is one-indexed
+
+        String expectedMessage = String.format(ViewCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                               personToBeViewed.getAsTextHidePrivate());
+        assertViewBehavior(new ViewCommand(targetVisibleIndex), addressBook, relevantPersons, expectedMessage);
+
+        expectedMessage = String.format(ViewAllCommand.MESSAGE_VIEW_PERSON_DETAILS,
+                                        personToBeViewed.getAsTextShowAll());
+        assertViewBehavior(new ViewAllCommand(targetVisibleIndex), addressBook, relevantPersons, expectedMessage);
+    }
+
+    /**
+     * Asserts that the Viewcommand and ViewAllcommand reports the given error for the given input
+     */
+    private static void assertViewError(AddressBook addressBook, List<ReadOnlyPerson> relevantPersons,
+                                        int targetVisibleIndex, String expectedMessage) {
+        assertViewBehavior(new ViewCommand(targetVisibleIndex), addressBook, relevantPersons, expectedMessage);
+        assertViewBehavior(new ViewAllCommand(targetVisibleIndex), addressBook, relevantPersons, expectedMessage);
+    }
+
+    /**
+     * Asserts that the ViewCommand and ViewAllCommand show given feedback information
```

Comment:
It does not just check for feedback information. You may wish to enumerate the list of things it checks for, e.g.
```
Checks that viewCommand exhibits the correct view command behavior, namely:

1. The feedback message of the CommandResult it returns matches expectedMessage
2. The CommandResult it returns has no relevant persons.
3. ...
```